dashboard/test/controllers/redirect_proxy_controller_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'webmock/minitest'
+WebMock.disable_net_connect!(:allow_localhost => true)
+require_relative '../../../shared/test/spy_newrelic_agent'
+require 'test_helper'
+
+class RedirectProxyControllerTest < ActionController::TestCase
+  LONG_URI = 'https://studio.code.org/s/course1/stage/3/puzzle/2'
+
+  test "should successfully get a redirect for url in whitelist" do
+    short_uri = 'http://bit.ly/2gPJmTQ'
```

Comment:
Note, this set of tests explicitly disables network access (such that any attempts to use it will result in failures - something we depend on). For these bit.ly URLs, we stub the web request as needed such that there isnt actually a network call.

In general, I think our tests use VCR or other means such that we don't need network access (tho I can't say with full confidence we do that across the board).